src/libpmemfile-posix/internal.h

```diff
@@ -49,27 +49,32 @@ pmemfile_tx_abort(int err)
 
 /*
  * The size of data allocated for each block is a positive integer multiple
- * of FILE_PAGE_SIZE.
+ * of BLOCK_ALIGNMENT.
  *
  * XXX: The current code can read from / write to blocks with any positive size,
  * any offset alignment, so this information doesn't necessarily have to be
  * part of the on-media layout.
  * But later the code might (probably will) depend on this.
  */
-#define FILE_PAGE_SIZE ((size_t)0x1000)
+#define MIN_BLOCK_SIZE ((size_t)0x1000)
 
-#define MAX_BLOCK_SIZE (UINT32_MAX - (UINT32_MAX % FILE_PAGE_SIZE))
+#define BLOCK_ALIGNMENT ((size_t)0x1000)
+
+COMPILE_ERROR_ON(MIN_BLOCK_SIZE < BLOCK_ALIGNMENT);
```

Comment:
Done. I can't imagine anyone proposing (or even testing) MIN_BLOCK_SIZE to become 0.